robottelo/cli/factory.py

```diff
@@ -3351,8 +3351,15 @@ def configure_env_for_provision(org=None, loc=None):
         'organization-ids': org['id'],
     })
 
-    # Search for SmartProxy, and associate location
-    puppet_proxy = Proxy.info({'id': Proxy.list()[0]['id']})
+    # If more than one proxy exists, Search for default SmartProxy
+    # else get first, and associate location
+    proxy = Proxy.list()
```

Comment:
when doing provisioning here we mean to use the default proxy (eg we already know which proxy we should use), why not find it directly as it should always exist.
eg:
```python
puppet_proxy = Proxy.info({'name': settings.server.hostname })
```

Reply:
@ldjebran yea, that will also work. I updated the commit.Thanks :+1: